docs/Development.md

```diff
@@ -73,6 +73,47 @@ Within the Pelorus GitHub fork directory:
 
        To ensure the implementation's pull request is merged **after** the ADR's PR it is recommended to add `Depends on: #ADR_PULL_REQUEST_NUMBER` as the first comment on the implementation PRs. Please follow the [dpulls](https://www.dpulls.com/docs#/dependencies) for more information.
 
+## Python dependencies
+
+The project uses [Poetry](https://python-poetry.org/) to manage the Python dependencies
+
+### Add a Python dependency
+
+To add a Python dependency, run
+```
+poetry add package_name
```

Comment:
Include here and the following, so it's clear the update requirements is necessary, otherwise one may miss that step. Remove `### Update requirements files` after such change as it won't be relevant anymore.

```shell
$ poetry add package_name
$ make update-requirements
```